### ISSUE TITLE
CDTS V4.0.32 ExitScript issue when using a link with target="_blank"

### ISSUE THREAD

**Issue by MaximMartel** [wet-boew/cdts-sgdc#49]
Moved from GCCode : https://gccode.ssc-spc.gc.ca/iitb-dgiit/nw-ws/sgdc-cdts/issues/149

@StdGit

[CDTS_V4_0_32_ExitScript_issue_when_using_a_link_with__target___blank__.zip](https://github.com/wet-boew/cdts-sgdc/files/4149723/CDTS_V4_0_32_ExitScript_issue_when_using_a_link_with__target___blank__.zip)

**Comment by jonnorth** [comment]
Any Update on this? We need the ability to open links in a new window. 

Here are the details from attached zip file:
We are currently updating from CDTS version 4.0.29 to v4.0.32 and noticed an issue where links that have an attribute target=””_blank” no longer open in a new window/tab. 

This is for an MVC application using the latest version 2.02 of the GoC .NET Webtemplate. 
For example this is our settings emitted for the refFooter:
```javascript
    <!-- Write closure template -->
    <script type="text/javascript">
        document.write(wet.builder.refFooter({"cdnEnv":"prod","exitScript":true,"exitURL":"/TestFeatures/ExternalLink/","displayModal":true}));
    </script>
```
Example  html link:
````
<p>Here is an external link that will display the warning:<a href="http://www.pinkbike.com"  target="_blank">Link to External Page</a></p>
````

From the code it looks like some changes in the exitScript.js removed the code dealing with the target=”_blank” attribute?

Thanks

**Comment by StdGit** [comment]
Some information about this issue.

1-) Redirection to another site didn't work before, it's why we had a bug. It was handled manually by the redirection page with response.redirect. or other techniques. CDTS didn't manage it.

2-) 

> From the code it looks like some changes in the exitScript.js removed the code dealing with the target=”_blank” attribute?

No it doesn't. Only the redirection was implemented. The feature to be able to open the link into another window wasn't. You could have more information in the WEB about [ target="_blank" ](https://css-tricks.com/use-target_blank/)

3-) I believe exitscript.js could be modified to include the new feature for the next release.

4-) P.S. Exitscript.js is added to wet.js during the compliation of the soy file. So you need to compile it.

5-) Ilya.pak@hrsdc-rhdcc.gc.ca offered to fix it.

**Comment by jonnorth** [comment]
@ipaksc Do we have a date yet when fix this will be released? Thanks

**Comment by ipaksc** [comment]
@jonnorth I think in April...

**Comment by patricehernert** [comment]
@MaximMartel Could you confirm this issue is still valid for version 4.0.39 ? Thanks.

**Comment by StdGit** [comment]
Cela semble fonctionner, cependant nous aurions besoin d'un exemple vu que
l'exemple actuel utilise document.write et n'inclut pas un exemple avec target_blank ce qui peut occasionner des problèmes dans certaines situations/fureteurs.

This seems to work. However we would need an example as
the current example uses document.write and does not include an example with target_blank which can cause problems in certain situations/browsers

https://developers.google.com/web/updates/2016/08/removing-document-write
https://support.google.com/chrome/thread/68124424?hl=en

**Comment by ahmad-shahid** [comment]
Fixed/Tested as part of #198/#199.
